### PR TITLE
[FHIR] Use app.kubernetes.io-scoped labels

### PR DESCRIPTION
**Pull request by chgl**
This replaces the previous labels with common/recommended labels: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels

### CODE REVIEW

File: charts/ibm-fhir-server/values.yaml [hunk 1]
```diff
@@ -5,6 +5,11 @@ fullnameOverride:
 # -- Optional override for chart name portion of the created kube resources
 nameOverride:
 
+# -- Extra labels to apply to the created kube resources
+extraLabels:
+  {}
+  # team: fhir
```
Comment (lmsurpre) [inline charts/ibm-fhir-server/values.yaml:11]: I guess this is an example value?  I'm not sure it has much value here.  I'm not sure the example is even needed, but if we want to keep it we could make it part of the documentation for the field.
```suggestion
```

**Comment by lmsurpre** [comment]
I find it a little odd that the kubernetes docs say these are the recommended labels but then their own documentation for labels/selects still uses the simpler unprefixed "app" label.
I do like the new helper for the the common label annotations.

**Review by lmsurpre — APPROVED** [review]
LGTM

**Review by michaelwschroeder — APPROVED** [review]
LGTM

**Comment by lmsurpre** [comment]
I didn't realize it at the time, but this is a backwards-breaking change.  If you try upgrading an existing install you will get something like this :
> Error: UPGRADE FAILED: cannot patch "ibm-fhir-server" with kind Deployment: Deployment.apps "ibm-fhir-server" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"server", "app.kubernetes.io/instance":"ibm-fhir-server", "app.kubernetes.io/name":"ibm-fhir-server"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable

**Comment by chgl** [comment]
Hm, I wasn't sure at the time either to be honest. I assumed chart testing's upgrade test would catch this: https://github.com/Alvearie/alvearie-helm/runs/3723826708?check_suite_focus=true#step:7:30

And

> If upgrade (--upgrade) is true, then this command will validate that 'helm test' passes for the following upgrade paths:

>previous chart revision => current chart version (if non-breaking SemVer change)
>current chart version => current chart version

https://github.com/helm/chart-testing/blob/main/doc/ct_install.md

I'll have to double check why it didn't get caught 🤔

**Pull request merged**